app/src/main/java/org/gdg/frisbee/android/appwidget/UpcomingEventWidgetProvider.java

```diff
@@ -123,7 +123,18 @@ public void onNotFound(String key) {
                 });
 
         }
-
+        private Event getNextEvent(List<Event> listEvents) {
+            if (listEvents == null || listEvents.size() == 0) {
+                return null;
+            }
+            Event ret = listEvents.get(0);
```

Comment:
We are looking for the closest future one to today. I am confused because you are not using today's date in this algorithm. 
I am pretty sure that the list is sorted once we got the list from the backend. The problem is that we separated the past and future events. When we did that separation, we didn't split the list into 2. We just changed the ordering. 
Maybe we really should need 2 separate lists. 

This method will be confusing to everybody in the future.